src/components/navbar/navbar.js

```diff
@@ -4,12 +4,34 @@ import { useAuth } from "../../context/AuthContext";
 import { useWishList} from "../../context/WishListContext";
 import { useCart } from "../../context/CartContext";
 import LogOutHandler from "../../services/logoutService";
+import { useState } from "react";
+import { useProduct } from "../../context/product-context";
+import { useFilter } from "../../context/filter-context";
+
 export default function Navbar() {
-    
-    const navigate=useNavigate;
+
+    const  {setSearchTerm}=useFilter();
+    const navigate=useNavigate();
+    const[inputKey,setInputKey]=useState("")
     const {auth:{isAuthenticated,user}}=useAuth()
     const {wishList}=useWishList()
     const {cart}=useCart()
+   
+   console.log(inputKey)
+   const searchHandler = () => {
+    if (setSearchTerm === "") {
+      return;
+    } else {
+      setSearchTerm(inputKey);
+      if (inputKey === "") {
+        return;
```

Comment:
you can check both conditions in one if condition itself

Reply:
for some purpose only i have used like this nikhil